src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -40,6 +43,21 @@ internal struct ReadStackFrame
         public bool IsProcessingEnumerable => IsEnumerable || IsPropertyEnumerable;
         public bool IsPropertyEnumerable => JsonPropertyInfo != null ? JsonPropertyInfo.ClassType == ClassType.Enumerable : false;
 
+        public bool IsProcessingProperty
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                if (JsonPropertyInfo == null || Skip())
+                {
+                    return false;
+                }
+
+                ClassType type = JsonPropertyInfo.ClassType;
+                return type == ClassType.Value || type == ClassType.Unknown;
```

Comment:
Seems like we call this property quite often (in the reader loop). Is there value in re-ordering the `ClassType` enum to have Unknown and Value be next to each other (Value = 1 instead of 2)? Then, it can be a single comparison.

https://github.com/dotnet/corefx/blob/648d0724c2b9224cace4b1e179fc20904fe27aa4/src/System.Text.Json/src/System/Text/Json/Serialization/ClassType.cs#L10-L17

Reply:
Are you suggesting checking for `<`?

Reply:
Yes. If Unknown = 0, Value = 1, you could do return `type <= ClassType.Value`.